From 9df008580feacd7e941b836f29d275395ef46ec6 Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Tue, 13 May 2014 09:15:51 -0400
Subject: [PATCH 116/117] RHEL 6: disable fstrim API (RHBZ#982979).

The combination of qemu, kernel & virtio-scsi in RHEL 6 is too old to
make fstrim work properly.
Thus, make the 'fstrim' feature always unavailable, and hide the
'fstrim' command from the documentation.
---
 daemon/fstrim.c           | 3 ++-
 generator/actions.ml      | 1 +
 perl/t/006-pod-coverage.t | 1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/daemon/fstrim.c b/daemon/fstrim.c
index cb844d1..b86f729 100644
--- a/daemon/fstrim.c
+++ b/daemon/fstrim.c
@@ -35,7 +35,8 @@ GUESTFSD_EXT_CMD(str_fstrim, fstrim);
 int
 optgroup_fstrim_available (void)
 {
-  return prog_exists (str_fstrim);
+  /* Disabled in RHEL 6, https://bugzilla.redhat.com/show_bug.cgi?id=982979 */
+  return 0;
 }
 
 /* Takes optional arguments, consult optargs_bitmask. */
diff --git a/generator/actions.ml b/generator/actions.ml
index 4fa2ae3..2f9c0ec 100644
--- a/generator/actions.ml
+++ b/generator/actions.ml
@@ -9728,6 +9728,7 @@ See also C<guestfs_available>, L<guestfs(3)/AVAILABILITY>." };
   { defaults with
     name = "fstrim";
     style = RErr, [Pathname "mountpoint"], [OInt64 "offset"; OInt64 "length"; OInt64 "minimumfreeextent"];
+    visibility = VDebug;
     proc_nr = Some 334;
     optional = Some "fstrim";
     shortdesc = "trim free space in a filesystem";
diff --git a/perl/t/006-pod-coverage.t b/perl/t/006-pod-coverage.t
index dd82ee8..ea149d5 100644
--- a/perl/t/006-pod-coverage.t
+++ b/perl/t/006-pod-coverage.t
@@ -26,5 +26,6 @@ all_pod_coverage_ok ({
         qr/^debug.*/,
         qr/^is_busy$/,
         qr/^internal.*/,
+        qr/^fstrim$/,
         ]
 });
-- 
2.5.0

